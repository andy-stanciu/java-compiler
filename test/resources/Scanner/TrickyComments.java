Here are a few tricky comments. There should be no errors in this file.
Comment 1
/**/
Comment 2
/***/
Comment 3
/*/*/
Comment 4
/*/*****/
Comment 5
/*/////////fgerhe***/
Comment 6
/************************
 *
 *
 *
 *
 *
 */
Comment 7
/**
 * ZZ_LEXSTATE[l] is the state in the DFA for the lexical state l
 * ZZ_LEXSTATE[l+1] is the state in the DFA for the lexical state l
 *                  at the beginning of a line
 * l is of the form l = 2*k, k a non negative integer
 */
Comment 8
/**
 * If String compaction is disabled, the bytes in {@code value} are
 * always encoded in UTF16.
 *
 * For methods with several possible implementation paths, when String
 * compaction is disabled, only one code path is taken.
 *
 * The instance field value is generally opaque to optimizing JIT
 * compilers. Therefore, in performance-sensitive place, an explicit
 * check of the static boolean {@code COMPACT_STRINGS} is done first
 * before checking the {@code coder} field since the static boolean
 * {@code COMPACT_STRINGS} would be constant folded away by an
 * optimizing JIT compiler. The idioms for these cases are as follows.
 *
 * For code such as:
 *
 *    if (coder == LATIN1) { ... }
 *
 * can be written more optimally as
 *
 *    if (coder() == LATIN1) { ... }
 *
 * or:
 *
 *    if (COMPACT_STRINGS && coder == LATIN1) { ... }
 *
 * An optimizing JIT compiler can fold the above conditional as:
 *
 *    COMPACT_STRINGS == true  => if (coder == LATIN1) { ... }
 *    COMPACT_STRINGS == false => if (false)           { ... }
 *
 * @implNote
 * The actual value for this field is injected by JVM. The static
 * initialization block is used to set the value here to communicate
 * that this static final field is not statically foldable, and to
 * avoid any possible circular dependency during vm initialization.
 */
Comment 9
/* /* /* /* ****** */
Comment 10
// (1)We never cache the "external" cs, the only benefit of creating
// an additional StringDe/Encoder object to wrap it is to share the
// de/encode() method. These SD/E objects are short-lived, the young-gen
// gc should be able to take care of them well. But the best approach
// is still not to generate them if not really necessary.
// (2)The defensive copy of the input byte/char[] has a big performance
// impact, as well as the outgoing result byte/char[]. Need to do the
// optimization check of (sm==null && classLoader0==null) for both.
Comment 11
///////////////////////////////////////////////
//////////////////////////@#*@$
//////!@*!^&@$
///////////!@)(
Comment 12
// / //  // / / / /// &!@ 23u23/&**
// /****** */
// /*&@&#@#*@#*/
// */**/*/*****/*/*/*/**/*/*//// / /// // /
Done
