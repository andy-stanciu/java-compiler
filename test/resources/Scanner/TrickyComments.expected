ID(Here) ID(are) ID(a) ID(few) ID(tricky) ID(comments) DOT ID(There) ID(should) ID(be) ID(no) ID(errors) ID(in) THIS ID(file) DOT ID(Comment) INT(1) ID(Comment) INT(2) ID(Comment) INT(3) ID(Comment) INT(4) ID(Comment) INT(5) ID(Comment) INT(6) ID(Comment) INT(7) ID(Comment) INT(8) ID(Comment) INT(9) ID(Comment) INT(10) ID(Comment) INT(11) ID(Comment) INT(12) ID(Done) 